portfolio/src/main/webapp/script.js

```diff
@@ -1,86 +1,253 @@
-
-    const textElement = document.createElement('span');
-    textElement.innerText = comment.text;
-
-    const deleteButtonElement = document.createElement('button');
-    deleteButtonElement.innerText = 'Delete';
-    deleteButtonElement.addEventListener('click', () => {
-        deleteComment(comment);
-        commentElement.remove();
-        fetch("/data?limit=10").then(loadComments()) ;
-    });
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const textElement = document.createElement('span');
+  textElement.innerText = comment.text;
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
+    commentElement.remove();
+    setTimeout(function() {
+      fetch("/data?limit=10").then(loadComments()) ;
+    }, 100); 
+  });
 
-    commentElement.appendChild(textElement);
-    commentElement.appendChild(deleteButtonElement);
-    return commentElement;
+  commentElement.appendChild(textElement);
+  commentElement.appendChild(deleteButtonElement);
+  return commentElement;
 }
 
-/** Tells the server to delete the comment. */
+/** Tells the server to delete the comment.*/
 function deleteComment(comment) {
-    const params = new URLSearchParams();
-    params.append('id', comment.id);
-    fetch('/delete-data', {method: 'POST', body: params});
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-data', {method: 'POST', body: params});
+}
+
+/** Load charts using callback functions. */
+google.charts.load('current', 
+  {packages:['corechart'], callback: 
+    function() { 
+      drawScoreChart();
+      drawMinChart();
+    }
+  });
+
+/**
+ * Creates line plot of raw data and corresponding 7-day moving average. To reduce
+ * code duplication, the function is called for both the sleep score and the deep
+ * sleep minutes charts. The variable "score" is a boolean representing which chart
+ * is being plotted. Depending on the truth value of "score", different data is loaded
+ * and the corresponding chart elements are filled.
+ */
+function drawChart(score){
+  fetch('/sleep-data').then(response => response.json())
+  .then((sleep) => {
+    const data = new google.visualization.DataTable();
+    data.addColumn('date', 'Day');
+    if (score) {
+      data.addColumn('number', 'Sleep Score');
+    }
+    else {
+      data.addColumn('number', 'Deep Sleep Minutes');
+    }
+    data.addColumn('number', 'Moving Average');
+    data.addColumn({type:'string', role:'annotation'});
+    data.addColumn({type:'string', role:'annotationText'}); 
+
+    /** Hard-coded specific dates to apply annotation text to certain portion of graph. */
+    Object.keys(sleep).forEach((day) => {
+      if (score) {
+        rawScore = sleep[day][0];
+        maScore = sleep[day][2];
+          if (day == "2020-05-30") {
```

Comment:
This file seem to use both double quotes and single quotes for literal strings. The JavaScript style guide seem prefer single quotes.

Reference: https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

Reply:
fixed! For the strings that had the single quote character, I used the template strings as suggested by the style guide.